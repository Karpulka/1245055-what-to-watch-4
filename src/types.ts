export interface Film {
  id: number;
  title: string;
  src: string;
  background: string;
  preview: string;
  genre: string;
  year: number;
  description: string;
  rating: number;
  voiceCount: number;
  director: string;
  actorList: Array<string>;
  runtime: number;
  video: string;
  fullVideo: string;
  isFavorite: boolean;
}

export interface Details {
  genre?: string;
  year?: number;
  director?: string;
  actorList?: Array<string>;
  runtime?: number;
}

export interface Overview {
  description?: string;
  rating?: number;
  voiceCount?: number;
  director?: string;
  actorList?: Array<string>;
}

export interface Review {
  id: number;
  user: string;
  rating: number;
  comment: string;
  date: string;
}

export interface Tab {
  title: string;
  value: Details | Overview | Review | number;
}

export interface ActiveItem {
  title: string;
  value: Overview;
}

export interface RatingDescriptionType {
  min: number;
  max?: number;
  title: string;
}
