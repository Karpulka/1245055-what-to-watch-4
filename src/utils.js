import moment from "moment";
import momentDurationFormatSetup from "moment-duration-format";
momentDurationFormatSetup(moment);

export const prepareFilmDuration = (time) => {
  return moment.duration(time, `minutes`).format(`h[h] m[m]`);
};

export const formatTimeLeft = (time) => {
  return moment.duration(time, `seconds`).format(`hh:mm:ss`);
};

export const prepareReviewDate = (date) => {
  return moment(date).format(`MMMM DD, YYYY`);
};

export const setNewObject = (a, b) => {
  return Object.assign({}, a, b);
};
