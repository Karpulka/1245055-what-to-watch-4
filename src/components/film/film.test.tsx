import * as React from "react";
import * as renderer from "react-test-renderer";
import Film from "./film";
import {StaticRouter} from "react-router-dom";
import {Film as FilmType} from "../../types";
import {noop} from "../../utils";

const film: FilmType = {
  id: 6,
  title: `Остров проклятых`,
  src: `/shutter-island.jpg`,
  background: `/shutter-island.jpg`,
  preview: `/shutter-island.jpg`,
  genre: `Mystery, Thriller`,
  year: 2010,
  video: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  fullVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  description: `<p>In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.</p>
                  <p>Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.</p>`,
  rating: 8.87,
  isFavorite: true,
  voiceCount: 240,
  director: `Wes Andreson`,
  runtime: 115,
  actorList: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`]
};

it(`Render Film`, () => {
  const tree = renderer
    .create(<StaticRouter><Film
      film={film}
      onFilmClick={noop}
      onFilmHover={noop}
      onFilmBlur={noop}
      isStartPlaying={false}
    /></StaticRouter>, {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
