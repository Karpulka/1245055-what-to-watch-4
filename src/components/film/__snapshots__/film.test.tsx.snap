// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Film 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
    onClick={[Function]}
  >
    <video />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/6"
      onClick={[Function]}
    >
      Остров проклятых
    </a>
  </h3>
</article>
`;
